docker/release-images.mjs

```diff
@@ -98,118 +99,186 @@ import { execSync } from "node:child_process";
-  if (!argValue) {
-    console.error(chalk.red(`ERROR: Missing required argument or environment variable: ${arg}`));
-    process.exit(1);
+async function installCrane() {
+  if (getEnvironment() === Environment.CI) {
+    console.log("installing crane automatically in CI");
+    await $`curl -sL https://github.com/google/go-containerregistry/releases/download/v0.15.1/go-containerregistry_Linux_x86_64.tar.gz > crane.tar.gz`;
+    const sha = (await $`shasum -a 256 ./crane.tar.gz | awk '{ print $1 }'`).toString().trim();
+    if (sha !== "d4710014a3bd135eb1d4a9142f509cfd61d2be242e5f5785788e404448a4f3f2") {
+      console.error(chalk.red(`ERROR: sha256 mismatch for crane.tar.gz got: ${sha}`));
+      process.exit(1);
+    }
+    await $`tar -xf crane.tar.gz`;
+    crane = "./crane";
+  } else {
+    if ((await $`command -v crane`.exitCode) !== 0) {
+      console.log(
+        chalk.red(
+          "ERROR: could not find crane binary in PATH - follow https://github.com/google/go-containerregistry/tree/main/cmd/crane#installation to install",
+        ),
+      );
+      process.exit(1);
+    }
+    crane = "crane";
   }
-  parsedArgs[arg] = argValue;
 }
 
-for (const arg of OPTIONAL_ARGS) {
-  const argValue = argv[arg.toLowerCase().replaceAll("_", "-")] ?? process.env[arg];
-  parsedArgs[arg] = argValue;
+// This is kinda gross but means we can just run the script directly and it will
+// work without running pnpm install before hand...
+async function lazyImports() {
+  await import("zx/globals");
+  toml = await import("toml");
+  core = await import("@actions/core");
 }
 
-let crane;
+const Environment = {
+  CI: "ci",
+  LOCAL: "local",
+  TEST: "test",
+};
 
-if (process.env.CI === "true") {
-  console.log("installing crane automatically in CI");
-  await $`curl -sL https://github.com/google/go-containerregistry/releases/download/v0.15.1/go-containerregistry_Linux_x86_64.tar.gz > crane.tar.gz`;
-  const sha = (await $`shasum -a 256 ./crane.tar.gz | awk '{ print $1 }'`).toString().trim();
-  if (sha !== "d4710014a3bd135eb1d4a9142f509cfd61d2be242e5f5785788e404448a4f3f2") {
-    console.error(chalk.red(`ERROR: sha256 mismatch for crane.tar.gz got: ${sha}`));
-    process.exit(1);
+
+function getEnvironment() {
+  if (process.env.CI === "true") {
+    return Environment.CI;
+  } else if (import.meta.jest !== undefined) {
+    return Environment.TEST;
+  } else {
+    return Environment.LOCAL;
   }
-  await $`tar -xf crane.tar.gz`;
-  crane = "./crane";
-} else {
-  if ((await $`command -v crane`.exitCode) !== 0) {
-    console.log(
-      chalk.red(
-        "ERROR: could not find crane binary in PATH - follow https://github.com/google/go-containerregistry/tree/main/cmd/crane#installation to install",
-      ),
-    );
-    process.exit(1);
+}
+
+function reportError(message, opts={throwOnFailure: false}) {
+  if (getEnvironment() === Environment.CI) {
+    core.setFailed(message);
```

Comment:
nice!

Reply:
Yeah, actually this is a reason why we probably wanna move all this kind of logic into a typescript action, then we can tag / deploy / test / use them anywhere with much more advanced functionality